model/task_lifecycle.go

```diff
@@ -387,19 +387,17 @@ func resetTask(ctx context.Context, taskId, caller string) error {
 	if t.IsPartOfDisplay() {
 		return errors.Errorf("cannot restart execution task '%s' because it is part of a display task", t.Id)
 	}
+	if err = checkUsersPatchTaskLimit(t.Requester, caller, true, *t); err != nil {
+		return errors.Wrap(err, "updating patch task limit for user")
+	}
 	if err = t.Archive(ctx); err != nil {
 		return errors.Wrap(err, "can't restart task because it can't be archived")
 	}
-
 	if err = MarkOneTaskReset(ctx, t, caller); err != nil {
 		return errors.WithStack(err)
 	}
 	event.LogTaskRestarted(t.Id, t.Execution, caller)
 
-	if err := t.ActivateTask(caller); err != nil {
```

Comment:
I noticed this is redundant code since `MarkOneTaskReset` already activates the task, removed